refactor(syntax): move line terminator related variables/methods from identifier to `line_terminator` module

Dunqing opened the pull request:
Now, we have a `line_terminator` module, so moving line terminator-related variables and methods to their own module would be better for maintainability.

Boshen commented:
### Merge activity

* **Dec 4, 7:17 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(syntax): move line terminator related variables/methods from identifier to `line_terminator` module (#16486)

Now, we have a `line_terminator` module, so moving line terminator-related variables and methods to their own module would be better for maintainability.
